### PR TITLE
Feature/signup with google

### PR DESCRIPTION
**Pull request by KseniaKoms**
add endpoint auth/google/signup

### CODE REVIEW

File: .development.env [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+MYSQL_HOST=localhost
+MYSQL_USER=root
+MYSQL_DB=test_freelancer
+MYSQL_PASSWORD=228394
+MYSQL_PORT=3306
+MYSQL_ROOT_PASSWORD=root
+CLIENT_HOST = 4200
+
+GOOGLE_CLIENT_ID = 253619542281-miag0ub2aorap933406d1vg3bbte7j8m.apps.googleusercontent.com
+GOOGLE_SECRET_KEY = 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
```
Comment (lyudad) [inline .development.env]: Снимок экрана 2022-11-15 в 19 58 37

Comment (KseniaKoms) [reply .development.env]: I`ve updated

File: src/modules/auth/services/auth.service.ts [hunk 2]
```diff
@@ -55,4 +58,25 @@ export class AuthService implements IAuthService {
...
+          `user with  ${email} is already registered`,
+          HttpStatus.CONFLICT,
+        );
+      }
+      const user = await this.userService.createUser(email);
+      return user;
+    } catch (error) {
+      throw new HttpException(`${error}`, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
```
Comment (lyudad) [inline src/modules/auth/services/auth.service.ts:84]: I think you have problem with eslint. this rule is not applied in your code
Снимок экрана 2022-11-15 в 20 29 24

Comment (KseniaKoms) [reply src/modules/auth/services/auth.service.ts:84]: added the rule

File: src/database/entities/users.entity.ts [hunk 3]
```diff
@@ -19,18 +19,21 @@ export class User {
   @Exclude()
   @Column({
     nullable: false,
```
Comment (lyudad) [inline src/database/entities/users.entity.ts]: it could be true if register with google

File: src/main.ts [hunk 4]
```diff
@@ -15,6 +15,11 @@ async function start() {
   const document = SwaggerModule.createDocument(app, config);
   SwaggerModule.setup('api', app, document);
   app.useGlobalPipes(new ValidationPipe());
+  app.enableCors({
+    allowedHeaders: ['content-type'],
+    origin: process.env.CLIENT_HOST,
+    credentials: true,
+  });
   await app.listen(PORT, () => console.log(`Server started on port ${PORT}`));
```
Comment (lyudad) [inline src/main.ts]: Снимок экрана 2022-11-16 в 22 37 07

File: src/modules/auth/controllers/auth.controller.ts [hunk 5]
```diff
@@ -40,4 +42,12 @@ export class AuthController {
   confirm(@Query(ValidationPipe) query: ConfirmAccountDto) {
     return this.authService.confirmEmail(query.id);
   }
+  @ApiBody({ type: TokenDto })
+  @ApiResponse({ status: 200, description: 'register with Google' })
+  @Post(Routes.SIGNUP_GOOGLE)
+  @UsePipes(ValidationPipe)
+  @UseInterceptors(ClassSerializerInterceptor)
+  signupGoogle(@Body('token') token: string) {
```
Comment (lyudad) [inline src/modules/auth/controllers/auth.controller.ts]: Снимок экрана 2022-11-16 в 22 33 47

File: src/database/entities/users.entity.ts (hunk 3 above)
Comment (KseniaKoms) [reply src/database/entities/users.entity.ts]: update

File: src/main.ts (hunk 4 above)
Comment (KseniaKoms) [reply src/main.ts]: update

File: src/modules/auth/controllers/auth.controller.ts (hunk 5 above)
Comment (KseniaKoms) [reply src/modules/auth/controllers/auth.controller.ts]: update

**Review by lyudad — APPROVED** [review]

**Pull request merged**